dataset/models/tf/layers/conv.py

```diff
@@ -26,7 +17,13 @@ def conv(inputs, *args, **kwargs):
     `tf.layers.conv3d <https://www.tensorflow.org/api_docs/python/tf/layers/conv3d>`_
     """
     dim = inputs.shape.ndims - 2
-    layer_fn = _CONV_LAYERS[dim]
+    conv_layers = {
+        1: tf.layers.conv1d,
+        2: tf.layers.conv2d,
+        3: tf.layers.conv3d
+    }
```

Comment:
There is no point in creating this dict in every call. Create it once when the module is initialized or don't create it at all (use `if` instead)